metricflow/specs/specs.py

```diff
@@ -126,14 +126,15 @@ class MetadataSpec(InstanceSpec):
     """A specification for a specification that is built during the dataflow plan and not defined in config."""
 
     element_name: str
+    agg_type: Optional[AggregationType] = None
 
     @property
     def qualified_name(self) -> str:  # noqa: D
-        return self.element_name
+        return f"{self.element_name}{DUNDER}{self.agg_type.value}" if self.agg_type else self.element_name
```

Comment:
@plypaul - as we discussed, this is an example of where we want to have structured annotations. The plan is to add an better abstraction to the spec interface so we can unify this with the similar thing we're doing with non-additive time dimensions, and then build out support for auto-aliasing metric offsets on top of the same set of spec interfaces.